code/modules/reagents/chemistry/machinery/chem_dispenser.dm

```diff
@@ -228,7 +228,7 @@ obj/machinery/chem_dispenser/update_icon()
 				if(beaker && dispensable_reagents.Find(r_id)) // but since we verify we have the reagent, it'll be fine
 					var/datum/reagents/R = beaker.reagents
 					var/free = R.maximum_volume - R.total_volume
-					var/actual = min(round(chemicals_to_dispense[key], res), (cell.charge * powerefficiency)*10, free)
+					var/actual = min(max(chemicals_to_dispense[key], res), (cell.charge * powerefficiency)*10, free)
```

Comment:
Wait so instead of macros giving you a "resolution" of 5,3,2,1 units per macro, you can only dispense 5,3,2,1 units max per click? So upgrading the machine makes it worse? I don't get it

Reply:
uuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuh

*deletes this*

Reply:
you still haven't deleted this

Reply:
I know I suck senpai ;_;